src/hotspot/os/linux/cgroupSubsystem_linux.hpp

```diff
@@ -69,148 +69,49 @@
+{                                                                                     \
+  bool is_ok;                                                                         \
+  is_ok = controller->read_string(filename, &retval);                                 \
+  if (!is_ok) {                                                                       \
+    log_trace(os, container)(log_string " failed: %d", OSCONTAINER_ERROR);            \
+    return nullptr;                                                                   \
+  }                                                                                   \
+  log_trace(os, container)(log_string " is: %s", retval);                             \
 }
 
-#define GET_CONTAINER_INFO_CPTR(return_type, subsystem, filename,         \
-                               logstring, scan_fmt, variable, bufsize)    \
-  char variable[bufsize];                                                 \
-{                                                                         \
-  int err;                                                                \
-  err = subsystem_file_line_contents(subsystem,                           \
-                                     filename,                            \
-                                     nullptr,                             \
-                                     scan_fmt,                            \
-                                     variable);                           \
-  if (err != 0)                                                           \
-    return (return_type) nullptr;                                         \
-                                                                          \
-  log_trace(os, container)(logstring, variable);                          \
-}
 
-#define GET_CONTAINER_INFO_LINE(return_type, controller, filename,        \
-                           matchline, logstring, scan_fmt, variable)      \
-  return_type variable;                                                   \
-{                                                                         \
-  int err;                                                                \
-  err = subsystem_file_line_contents(controller,                          \
-                                filename,                                 \
-                                matchline,                                \
-                                scan_fmt,                                 \
-                                &variable);                               \
-  if (err != 0)                                                           \
-    return (return_type) OSCONTAINER_ERROR;                               \
-                                                                          \
-  log_trace(os, container)(logstring, variable);                          \
-}
+enum TupleValue { FIRST, SECOND };
```

Comment:
Suggestion/nit: Enum class + `First`, `Second`. Your choice to keep it as is, however.

Reply:
I really want a more descriptive name and/or comment for this.